optuna/storages/_journal/storage.py

```diff
@@ -93,8 +98,23 @@ def __init__(self, log_storage: BaseJournalLogStorage) -> None:
         self._replay_result = JournalStorageReplayResult(self._worker_id_prefix)
 
         with self._thread_lock:
+            if isinstance(self._backend, BaseJournalLogSnapshot):
+                self._backend.load_snapshot(self.restore_replay_result)
             self._sync_with_backend()
 
+    def restore_replay_result(self, snapshot: bytes) -> None:
+        try:
+            r: Optional[JournalStorageReplayResult] = pickle.loads(snapshot)
+        except Exception as e:
```

Comment:
Do you think `pickle.UnpicklingError` is enough to handle?
https://docs.python.org/3/library/pickle.html#pickle.UnpicklingError

```suggestion
        except pickle.UnpicklingError as e:
```

Reply:
We need to use `UnpicklingError` or `KeyError` here. See https://github.com/optuna/optuna/actions/runs/3417004183/jobs/5687697387